src/gui/editors/PianoRoll.cpp

```diff
@@ -1028,8 +1029,7 @@ void PianoRoll::drawNoteRect( QPainter & p, int x, int y,
 			QString noteKeyString = getNoteString(n->key());
 
 			QFont noteFont(p.font());
-			noteFont.setPixelSize(noteTextHeight);
-			QFontMetrics fontMetrics(noteFont);
+			QFontMetrics fontMetrics(adjustedToPixelSize(noteFont, noteTextHeight));
```

Comment:
Sorry, only noticed this one now. I think it has to be changed to the following:
```
QFont noteFont(p.font());
noteFont = adjustedToPixelSize(noteFont, noteTextHeight);
QFontMetrics fontMetrics(noteFont);
```
Otherwise the font metrics will work on a different font than the one that is later use to actually paint things, i.e. things will go out of sync. That's because, as noted before, `adjustedToPixelSize` returns an adjusted copy of the font.

The whole comment assumes that `p.setFont(noteFont)` is called at one place.

Reply:
Fixed it, but the screenshot you linked below is unrelated. Fixed that too.